test/webrtc_test.rb

```diff
@@ -0,0 +1,202 @@
+# frozen_string_literal: true
+
+using_task_library "comms_webrtc"
+import_types_from "iodrivers_base"
+import_types_from "base"
+
+describe OroGen.comms_webrtc.Task do
+    run_live
+
+    attr_reader :task1, :task2
+
+    run_pid = nil
+    before do
+        time = Time.now
+        timeout = Time.now + 1
+        while time < timeout
+            begin
+                run_pid = Process.spawn("rustysignal", "127.0.0.1:3012")
+                break
+            rescue Errno::ECONNREFUSED
+                sleep(0.01)
+            end
+            time = Time.now
+        end
+    end
+
+    after do
+        unless run_pid.nil?
+            Process.kill("KILL", run_pid)
+            Process.wait
+        end
+    end
+
+    def deploy(task_a, task_b, time_out, server_name)
+        syskit_stub_conf(OroGen.comms_webrtc.Task, task_a.to_s, task_b.to_s)
+
+        @task1 = syskit_deploy(
+            OroGen.comms_webrtc.Task.with_conf(task_a.to_s).deployed_as(task_a.to_s)
+        )
+        @task1.properties.data_channel_time_out = time_out
+        @task1.properties.peer_connection_time_out = time_out
+        @task1.properties.wait_remote_peer_time_out = time_out
+        @task1.properties.websocket_open_time_out = time_out
+        @task1.properties.stun_server = "stun:stun.l.google.com:19302"
+        @task1.properties.local_peer_id = task_a.to_s
+        @task1.properties.remote_peer_id = task_b.to_s
+        @task1.properties.data_channel_label = "test_data_label"
+        @task1.properties.signaling_server_name = server_name
+
+        @task2 = syskit_deploy(
+            OroGen.comms_webrtc.Task.with_conf(task_b.to_s).deployed_as(task_b.to_s)
+        )
+        @task2.properties.data_channel_time_out = time_out
+        @task2.properties.peer_connection_time_out = time_out
+        @task2.properties.wait_remote_peer_time_out = time_out
+        @task2.properties.websocket_open_time_out = time_out
+        @task2.properties.stun_server = "stun:stun.l.google.com:19302"
+        @task2.properties.local_peer_id = task_b.to_s
+        @task2.properties.remote_peer_id = ""
+        @task2.properties.data_channel_label = "test_data_label"
+        @task2.properties.signaling_server_name = server_name
+    end
+
+    def configure_and_start()
+        syskit_configure(task1)
+        syskit_configure(task2)
+
+        expect_execution do
+            task1.start!
+            task2.start!
+        end.to do
+            emit task1.start_event
+            emit task2.start_event
+        end
+    end
+
+    it "emits start only after the data channel is fully established" do
+        deploy("task_a", "task_b", Time.at(2), "127.0.0.1:3012")
+        configure_and_start
```

Comment:
You do have to test something. Like for instance value on the status port ?

I would suggest that you write a status sample at the end of the startHook and read and return it from `configure_and_start`.